src/katsdpcontroller/schemas/product_config.json.j2

```diff
@@ -603,6 +606,22 @@
                                 "additionalProperties": false,
                                 "required": ["src_streams", "int_time"]
                             }
+                        },
+                        {
+                            "if": {"properties": {"type": {"const": "gpucbf.tied_array_channelised_voltage"}}},
+                            "then": {
+                                "properties": {
+                                    "type": {},
+                                    "src_streams": {"$ref": "#/definitions/singleton"},
+                                    "src_pol": {"$ref": "#/definitions/positive_integer"},
```

Comment:
positive_integer is definitely wrong, since it disallows 0. It should actually be stricter and only allow 0 or 1. I'm rusty on the syntax, but I think something like
```suggestion
                                    "src_pol": {"type": { "enum": [0, 1] }},
```

Reply:
I'd originally uhm'd and ah'd between `positive_integer` and `positive_number`, but I figured the latter was more for float-types. I admittedly couldn't find anything online Re: syntax, but appreciate the heads-up - will do more digging to confirm your suggestion works.

Reply:
Didn't have to look too far to confirm syntax, there are existing uses in this file. One of particular interest/inspiration is around line 164, under `cbf.antenna_channelised_voltage`
* `"n_pols": {"type": "integer", "enum": [1, 2]},`

Ignoring the actual values, I see the `"integer"` field/identifier is optional; other usages simply follow
* `"field_name": { "type": {"enum": ["value1", "value2", ... ]} }`

I gather the inclusion of `integer` in the first excerpt/example meant the options had to start from 1 rather than 0. I think I'm going to stick with your original suggestion; please and thank you once again.

Reply:
Actually, the second usage is invalid syntax - perhaps you can correct them. The value of `type` should always be a string or a list of strings, each of which is a primitive type (see the spec [here](https://json-schema.org/draft-07/draft-handrews-json-schema-validation-01#rfc.section.6.1)).

For `n_pols`, the `type` is mostly redundant, because if a value equals 1 or 2 it must be an integer. However, I think the Python json_schema package may use it to disallow `1.0` (JSON itself doesn't distinguish between integers and integral floats), so it probably doesn't hurt to have it. So I think the correct thing to use here would be
```json
"src_pol": {"type": "integer", "enum": [0, 1]}
```

Reply:
> Actually, the second usage is invalid syntax - perhaps you can correct them.

Ignore me - I see what's going on. It's valid syntax in its context: in the context, the `"type"` isn't a constraint (saying that a field must have a particular type), it's the *name* of a property in the document to be validated i.e. a stream must have a key called `"type"` and it's value must adhere to the `"enum"` restriction defined within.

Reply:
I'm going to try your suggestion (below) and poke sdp-jenkins to build it.
```JSON
"src_pol": {"type": "integer", "enum": [0, 1]}
```